### PR TITLE
fix(backend): Add'v' prefix in repo tags. Fixes #5954

### LINKED ISSUE #5954: [backend] missing 'v' prefix in repo tags
When trying to use the go client in github.com/kubeflow/pipelines/backend/api/go_client, I find that `go mod` can't fetch the corresponding tags in this github repo, because all tags defined are missing the 'v' prefix and are considered non-canonical tag formats.

E.g., tag '1.5.1' should be 'v1.5.1'.

### PR DESCRIPTION
**Pull request by NikeNano**
**Description of your changes:**

This PR fixes: https://github.com/kubeflow/pipelines/issues/5954. 

Related work is: 
- https://github.com/kubeflow/pipelines/pull/6025
- https://github.com/kubeflow/pipelines/pull/6034
- https://github.com/kubeflow/pipelines/pull/5974

See comment: https://github.com/kubeflow/pipelines/issues/5954#issuecomment-879558287

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -25,11 +25,11 @@ additionally.
...
 Releases are tagged with tags like `X.Y.Z`, e.g. `1.0.2`. A special format like
 `1.0.0-rc.2` is a pre-release. It is the second release candidate before
-releasing the final `1.0.0`.
+releasing the final `1.0.0`. For git tags a prefix, `v` will be added to support [golang modules](https://golang.org/ref/mod#versions).
 
 A release branch has the name structure `release-X.Y` where `X.Y` stands for the
-minor version. Releases like `X.Y.Z` will all be released from the branch
-`release-X.Y`.
+minor version. Releases like `vX.Y.Z` will all be released from the branch
+`vrelease-X.Y`.
```
Comment (Bobgy) [inline RELEASE.md:32]: nit: typo?

**Review by Bobgy — COMMENTED** [review]
/lgtm

File: RELEASE.md (hunk 1 above)
Comment (NikeNano) [reply RELEASE.md]: Yes, fixed now.

**Comment by Bobgy** [comment]
/hold
Actually, there are several places that refer to GitHub tags, let me try to find them

**Comment by NikeNano** [comment]
> /hold
> Actually, there are several places that refer to GitHub tags, let me try to find them

You mean in the docs or in the bash scripts?

**Comment by Bobgy** [comment]
I mean in bash scrpts

**Comment by NikeNano** [comment]
> I mean in bash scrpts

I cant find anything that I see should effect the github tag, so will wait for your feedback. Thanks :)

**Pull request closed without merging**